Symfony/CS/Fixer/Symfony/MethodSeparationFixer.php

```diff
@@ -156,6 +159,42 @@ private function fixSpaceAboveMethod(Tokens $tokens, $classStart, $methodIndex)
             }
         }
 
+        // deal with comments above a method
+        if ($tokens[$nonWhiteAbove]->isGivenKind(T_COMMENT)) {
+            if ($firstMethodAttrIndex - $nonWhiteAbove === 1) {
+                // no white space found between comment and method start
+                $this->correctLineBreaks($tokens, $nonWhiteAbove, $firstMethodAttrIndex, 1);
+
+                return;
+            }
+
+            // $tokens[$nonWhiteAbove+1] is always a white space token here
+            $lineBreakCount = substr_count($tokens[$nonWhiteAbove + 1]->getContent(), "\n");
+            if ($lineBreakCount > 1) {
+                // more than one line break, always bring it back to 2 line breaks between the method start and what is above it
+                $this->correctLineBreaks($tokens, $nonWhiteAbove, $firstMethodAttrIndex, 2);
+
+                return;
```

Comment:
https://github.com/FriendsOfPHP/PHP-CS-Fixer/pull/1426#discussion_r45794736

>The if{if-else{if-else{if-else}}}-construct is hard to read ;)

https://github.com/FriendsOfPHP/PHP-CS-Fixer/pull/1497#discussion_r46095848
;)

anyhow, I'm always happy to refactor ^-^